### PR TITLE
[SPARK-17778][Tests]Mock SparkContext to reduce memory usage of BlockManagerSuite

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Mock SparkContext to reduce memory usage of BlockManagerSuite
## How was this patch tested?

Jenkins

### CODE REVIEW

**Comment by zsxwing** [comment]
retest this please

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala [hunk 1]
```diff
@@ -107,7 +107,8 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     rpcEnv = RpcEnv.create("test", "localhost", 0, conf, securityMgr)
     conf.set("spark.driver.port", rpcEnv.address.port.toString)
 
-    sc = new SparkContext("local", "test", conf)
+    sc = mock(classOf[SparkContext])
```
Comment (rxin) [inline core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala]: might want to comment on the reason for this change too

**Comment by srowen** [comment]
Does this not defeat some of the purpose of testing, if it isn't using an actual SparkContext?

**Comment by andrewor14** [comment]
I think that's OK. This is supposed to be a unit test for the BlockManager, not how BlockManager interacts with the rest of the system. LGTM

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala (hunk 1 above)
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala]: > might want to comment on the reason for this change too

Added

**Comment by zsxwing** [comment]
> Does this not defeat some of the purpose of testing, if it isn't using an actual SparkContext?

The only reason we need to create a SparkContext is to initialize LiveListenerBus. There was no SparkContext in this file before we added a configuration to LiveListenerBus.

**Comment by markhamstra** [comment]
@zsxwing `build/mvn -U -Pyarn -Phadoop-2.7 -Pkinesis-asl -Phive -Phive-thriftserver -Dpyspark -Dsparkr test` completely succeeded on one of my machines where it was previously failing.

**Comment by zsxwing** [comment]
Thanks! Merging to master and 2.0

**Pull request closed without merging**